code_submitter/auth.py

```diff
@@ -199,3 +201,45 @@ def __init__(self, data: List[NemesisUserInfo] = DEFAULT) -> None:
 
     async def load_user(self, username: str, password: str) -> NemesisUserInfo:
         return self.data[username]
+
+
+class FileBackend(BasicAuthBackend):
+    """
+    Authentication backend which stores credentials in a YAML file.
+
+    Credentials are stored in the format `TLA: password`.
+    """
+
+    UNKNOWN_USER_MESSAGE = "Username or password is incorrect"
+    BLUESHIRT_TEAM = "SRX"
+
+    def __init__(
+        self,
+        *,
+        path: str,
+    ) -> None:
+        with open(path) as f:
+            self.credentials = cast(Dict[str, str], YAML(typ="safe").load(f))
+
+    def get_scopes(self, username: str) -> List[str]:
+        scopes = ['authenticated']
+
+        if username == self.BLUESHIRT_TEAM:
+            scopes.append('blueshirt')
```

Comment:
Maybe worth pulling `'blueshirt'` out to a constant now we've got a few instances of it?